### ISSUE TITLE
[Feature Request] Increase of Resolution and Texture Filtering

### ISSUE THREAD

**Issue by Papermanzero** [Yabause/yabause#16]
Hi Yabause team,
your emulator is meanwhile in my opinion the best saturn emulator.
However I have I request concerning the upscaling and filtering.
Most of the modern emulators like mupen64plus, pcsx-reloaded, dolphin, pcsx2 etc. can increase the internal system resolution to a certain value (x2, x2.5, x3...). This would be convenient for several reasons. The OpenGL renderer already increases the internal resolution to a certain value. However some persons tend to prefer to use the original resolution and other would like to increase the resolution even higher. The demand of both groups can be satisfied, if you enable a free selection of the internal resolution scaling value.
Furthermore a filter for the textures is really necessary to play games on higher resolutions especially on HDTVs. A bilinear, 2xSai, hq4x etc. would therefore be great. At least a bilinear filter would already improve the textures a lot. Alternatively you can implement shaders, which will take over the filtering. 
Maybe the filters of the other open source emulators can be supportive.

**Comment by ADormant** [comment]
Newer OpenGL would allow for a better filtering but even with the old OpenGL it should be doable.

https://www.opengl.org/wiki/Sampler_Object
https://www.opengl.org/registry/specs/EXT/texture_filter_anisotropic.txt
https://www.opengl.org/wiki/GLAPI/glTexParameter
https://www.opengl.org/wiki/GLAPI/glTexImage2D

https://github.com/Yabause/yabause/issues/17

**Comment by ADormant** [comment]
Xbrz is a good good filter for 2D. 
http://sourceforge.net/projects/xbrz/files/?source=navbar
https://github.com/hrydgard/ppsspp/tree/master/ext/xbrz
https://github.com/devinacker/bsnes-plus/pull/27/files
@Guillaumito 
@cyberwarriorx

**Comment by paul-met** [comment]
Current bilinear filtering does not work on win 10 x64.

**Comment by rdanbrook** [comment]
@paul-met, that's possibly because the bilinear filter is only for the software renderer. There is currently no filtering done in the OpenGL high-level renderer, but maybe someone like me can add it eventually ;).

**Comment by paul-met** [comment]
But software rendering no longer works for me. Just black screen. I thought that it was abandoned.

**Comment by rdanbrook** [comment]
Software rendering does actually use OpenGL, but only to blit a software-rendered pixel array. You can use the OpenGL renderer though? That doesn't really add up, because if the OpenGL renderer works then you have OpenGL 3.x capabilities, which is what the software renderer now uses.

**Comment by paul-met** [comment]
It is clear now. If you put a Software interface by default, it works. But if during the game his switch to OpenGL, and back to Software, the black screen appears.

**Comment by rdanbrook** [comment]
Yes, I have noticed that as well but have not gotten familiar enough with the codebase to diagnose/fix it.

**Comment by wingedonezero** [comment]
I would really like to see external shaders or filtering added yabause is the best but the only thing its lacking is this.Which is common in most other emulators at the current time.With resolution increasing all the time this would make the emulator more future proof as well.
